src/java/org/apache/cassandra/utils/Throwables.java

```diff
@@ -189,13 +189,19 @@ public static Throwable perform(Throwable accumulate, String filePath, FileOpTyp
     }
 
     public static void closeAndAddSuppressed(@Nonnull Throwable t, AutoCloseable... closeables)
+    {
+        closeAndAddSuppressed(t, Arrays.asList(closeables));
+    }
+
+    public static void closeAndAddSuppressed(@Nonnull Throwable t, Iterable<AutoCloseable> closeables)
     {
         Preconditions.checkNotNull(t);
         for (AutoCloseable closeable : closeables)
         {
             try
             {
-                closeable.close();
+                if (closeable != null)
```

Comment:
This is a new method. `FileUtils.close` just throws while here we are adding suppressed exceptions. Actually this works very similar to try..catch with resources, where only non-null resources are closed and no NPE is thrown. If you think I should move it to `FileUtils` I'm ok with that.

Reply:
I guess the comment applies to the method below then, but if we don't change it there it would be unexpected to skip here &mdash; unless we make the naming very clear about that difference.

Reply:
9ffff01f36d271686bac16b932b57c7d918a355d as well

Reply:
looks good